docs/sources/mimir/release-notes/v2.11.md

```diff
@@ -0,0 +1,92 @@
+    - `-blocks-storage.bucket-store.metadata-cache.memcached.write-buffer-size-bytes`
+  - For the query frontend:
+    - `-query-frontend.results-cache.memcached.read-buffer-size-bytes`
+    - `-query-frontend.results-cache.memcached.write-buffer-size-bytes`
+  - For the ruler storage:
+    - `-ruler-storage.cache.memcached.read-buffer-size-bytes`
+    - `-ruler-storage.cache.memcached.write-buffer-size-bytes`
+- **Configure the number of long-living workers used to process gRPC requests.** This can decrease CPU usage by reducing the number of stack allocations. Configure this feature by using the `-server.grpc.num-workers` CLI flag.
+- **Enforce a limit in bytes on the `PostingsForMatchers()` cache used by ingesters.** This limit can be configured via the `-blocks-storage.tsdb.head-postings-for-matchers-cache-max-bytes` and `-blocks-storage.tsdb.block-postings-for-matchers-cache-max-bytes` CLI flags.
+- ** Pre-allocate the pool of workers in the distributor that are used to send push requests to ingesters.** This can decrease CPU usage by reducing the number of stack allocations. You can enable this feature by using the `-distributor.reusable-ingester-push-worker` flag.
+
+### Helm chart improvements
+
+The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [Grafana Mimir Helm chart documentation](/docs/helm-charts/mimir-distributed/latest/).
+
+## Important changes
+
+In Grafana Mimir 2.11 the following behavior has changed:
+
+- The distributor `Push()` endpoint will now return the following gRPC codes instead of HTTP status codes:
+  - `http.StatusAccepted` (202) code is replaced with `codes.AlreadyExists`.
+  - `http.BadRequest` (400) code is replaced with `codes.FailedPrecondition`.
+  - `http.StatusTooManyRequests` (429) and the non-standard `529` (The service is overloaded) codes are replaced with `codes.ResourceExhausted`.
+- The utilization-based read path limiter now operates on Go heap size instead of RSS from the Linux proc file system.
+
+The following configuration options had been previously deprecated and are removed in Grafana Mimir 2.11:
+
+- The CLI flags `-querier.iterators` and `-query.batch-iterators`.
+- The CLI flag `-blocks-storage.bucket-store.bucket-index.enabled`.
+- The CLI flags `-blocks-storage.bucket-store.chunk-pool-min-bucket-size-bytes`, `-blocks-storage.bucket-store.chunk-pool-max-bucket-size-bytes` and `-blocks-storage.bucket-store.max-chunk-pool-bytes`.
+
+The following configuration options are deprecated and will be removed in Grafana Mimir 2.13:
+
+- The CLI flag `-log.buffered`; this is now the default behavior.
+
+The following metrics are removed:
+
+- `cortex_query_frontend_workers_enqueued_requests_total`; use `cortex_query_frontend_enqueue_duration_seconds_count` instead.
+
+The following experimental options are marked as stable:
+
+- The CLI flag `querier.prefer-streaming-chunks-from-ingesters`.
+- The CLI flag `querier.minimize-ingester-requests`.
```

Comment:
These flags are still considered experimental - the default value has changed to enable these features by default in this release though.

The plan is to remove the flags entirely in a future release, so these features are just always enabled by default.

Reply:
Got it, I'll remove them from here.